python/GafferUI/SpreadsheetUI/_RowsPlugValueWidget.py

```diff
@@ -403,18 +421,19 @@ def __updateDefaultRowVisibility( self ) :
 		if visible is None :
 			visible = True
 		self.__defaultLabel.setVisible( visible )
-		## \todo We shouldn't really be reaching into the protected
-		# `_qtWidget()` implementation here. Soon enough we will want
-		# to implement searching/filtering of rows, and when we implement
-		# that we should do it via a simple abstraction on `_PlugTableView`
-		# and use that here too. Perhaps just `setRowFilter( matchPattern )`
-		# would do the trick?
-		self.__defaultTable._qtWidget().setRowHidden( 0, not visible )
+		self.__defaultTable.setRowFilter( "" if visible else "__hideDefaultRow__" )
+
+	def __updatePatternWidgetWidth( self ) :
+
+		width = Gaffer.Metadata.value( self.getPlug().defaultRow(), "spreadsheet:rowNameWidth" ) or GafferUI.PlugWidget.labelWidth()
+		self.__patternWidget._qtWidget().setFixedWidth( width + 20 )
```

Comment:
It seems a pity that we have to duplicate this logic here. I experimented a bit with removing the expanding size policy from the text widget, and that seemed like it _might_ be viable, but had some annoying offset I didn't have time to figure out.